### PR TITLE
add vector macropad

### PR DESCRIPTION
**Pull request by daviddoan**
Adding macropad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/arrayperipherals/vector/config.h
Comment (zvecr) [inline keyboards/arrayperipherals/vector/config.h:1]: Please remove this file

File: keyboards/arrayperipherals/vector/rules.mk
Comment (zvecr) [inline keyboards/arrayperipherals/vector/rules.mk:1]: Please remove this file

File: keyboards/arrayperipherals/vector/vector.c
Comment (zvecr) [inline keyboards/arrayperipherals/vector/vector.c:1]: Please remove this file

File: keyboards/arrayperipherals/vector/vector.h
Comment (zvecr) [inline keyboards/arrayperipherals/vector/vector.h:1]: Please remove this file

File: keyboards/arrayperipherals/vector/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     KC_MPRV, KC_MPLY, KC_MNXT, KC_AUDIO_MUTE, TG(1)
+   ),
+
+  [1] = LAYOUT_ortho_1x5(
+     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, TG(1)
+  ),
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```
Comment (zvecr) [inline keyboards/arrayperipherals/vector/keymaps/default/keymap.c:32]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
```

File: keyboards/arrayperipherals/vector/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = { //button closest to usb is first
```
Comment (zvecr) [inline keyboards/arrayperipherals/vector/keymaps/default/keymap.c:20]:
```suggestion
const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```

File: keyboards/arrayperipherals/vector/keyboard.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    "keyboard_name": "The Vector",
+    "maintainer": "daviddoan",
+    "bootloader": "atmel-dfu",
+    "bootmagic": {
+        "enabled": true,
+        "matrix": [0, 0]
+    },
+    "dynamic_keymap": {
+        "layer_count": 2
+    },
```
Comment (zvecr) [inline keyboards/arrayperipherals/vector/keyboard.json:12]:
```suggestion
```

File: keyboards/arrayperipherals/vector/keyboard.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": true,
+        "encoder": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "raw": true,
+        "rgblight": false
```
Comment (zvecr) [inline keyboards/arrayperipherals/vector/keyboard.json:27]:
```suggestion
        "encoder": true,
        "extrakey": true,
        "mousekey": true,
        "nkro": true
```

File: keyboards/arrayperipherals/vector/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  
+  [0] = LAYOUT_ortho_1x5(
+     KC_MPRV, KC_MPLY, KC_MNXT, KC_AUDIO_MUTE, TG(1)
+   ),
+
+  [1] = LAYOUT_ortho_1x5(
```
Comment (zvecr) [inline keyboards/arrayperipherals/vector/keymaps/default/keymap.c:26]:
```suggestion
  [0] = LAYOUT(
     KC_MPRV, KC_MPLY, KC_MNXT, KC_AUDIO_MUTE, TG(1)
   ),

  [1] = LAYOUT(
```

File: keyboards/arrayperipherals/vector/keyboard.json [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    },
+    "processor": "atmega32u4",
+    "url": "https://arrayperipherals.com",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x4F47",
+        "vid": "0x4152"
+    },
+    "layouts": {
+        "LAYOUT_ortho_1x5": {
```
Comment (zvecr) [inline keyboards/arrayperipherals/vector/keyboard.json:42]:
```suggestion
        "LAYOUT": {
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by daviddoan — COMMENTED** [review]
accepted all changes!

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/arrayperipherals/vector/keyboard.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        "vid": "0x4152"
+    },
+    "layouts": {
+        "LAYOUT": {
+            "layout": [
+                {"label": "k01", "matrix": [0, 0], "x": 0, "y": 0},
+                {"label": "k02", "matrix": [0, 1], "x": 1.25, "y": 0},
+                {"label": "k03", "matrix": [0, 2], "x": 2.25, "y": 0},
+                {"label": "k04", "matrix": [0, 3], "x": 3.25, "y": 0},
+                {"label": "k05", "matrix": [0, 4], "x": 4.5, "y": 0}
```
Comment (fauxpark) [inline keyboards/arrayperipherals/vector/keyboard.json:36]: Is the key spacing meant to look like this?
image
The image in the readme suggests more of an even spacing.

File: keyboards/arrayperipherals/vector/readme.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+A small (1x4) macropad with a rotary. [More info at arrayperipherals.com](https://www.arrayperipherals.com/)
+
+* Keyboard Maintainer: [David Doan](https://github.com/daviddoan)
+* Hardware Supported: Custom PCB with Atmega32u4 processor
+* Hardware Availability: [arrayperipherals.com](https://www.arrayperipherals.com/)
+
+Make example for this keyboard (after setting up your build environment):
+
+	make arrayperipherals/vector:default
```
Comment (fauxpark) [inline keyboards/arrayperipherals/vector/readme.md:13]:
```suggestion
    make arrayperipherals/vector:default
```

File: keyboards/arrayperipherals/vector/readme.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+Flashing example for this keyboard:
+
+    make arrayperipherals/vector:default:flash
+
```
Comment (fauxpark) [inline keyboards/arrayperipherals/vector/readme.md:18]:
```suggestion
```

**Review by fauxpark — APPROVED** [review]

File: keyboards/arrayperipherals/vector/keyboard.json (hunk 7 above)
Comment (daviddoan) [reply keyboards/arrayperipherals/vector/keyboard.json]: Wow thanks for the catch!

**Review by fauxpark — APPROVED** [review]

**Pull request merged**